src/main/java/eu/stamp/project/assertfixer/asserts/AssertFixer.java

```diff
@@ -96,7 +96,7 @@ public boolean matches(CtInvocation element) {
             final List<CtCatch> catches = testCaseToBeFix.getElements(new TypeFilter<>(CtCatch.class));
             if (!catches.isEmpty()) {
                 TryCatchFixer.fixTryCatchFailAssertion(spoon, testCaseToBeFix, failure, catches);
-            } else {
+            } else if (configuration.isGenTryCatch()) {
                 TryCatchFixer.addTryCatchFailAssertion(spoon, testCaseToBeFix, failure);
             }
             repairType = AssertFixerResult.RepairType.TryCatchRepair;
```

Comment:
Shouldn't you move up the repairType? And then you might need a specific repairType for a final else case no?